agent/cmd/start.go

```diff
@@ -170,8 +170,12 @@ func newStartCommand() *cobra.Command {
 			}
 
 			// Get a single or a list of subscriptions
-			subscriptions := viper.GetStringSlice(flagSubscriptions)
-			cfg.Subscriptions = subscriptions
+			subscriptions := viper.GetString(flagSubscriptions)
+			if subscriptions != "" {
+				cfg.Subscriptions = splitAndTrim(subscriptions)
+			} else {
+				cfg.Subscriptions = viper.GetStringSlice(flagSubscriptions)
```

Comment:
Maybe I'm missing something but I don't see how we could reach this condition without the `flagSubscriptions` flag being empty, and therefore subscriptions value at all.

Both `--subscriptions foo,bar` & `--subscriptions foo --subscriptions bar` will cause the _if_ statement to be true, and in the latter case the first value (foo) will be ignored. 

For reference, https://github.com/spf13/viper/issues/380.

Reply:
I was just doing what you did :grin: 
https://github.com/sensu/sensu-go/blame/master/agent/cmd/start.go#L164

I'm not sure why it would drop the first value foo, but I think you're right about the if statement never being false.

Reply:
"Do what I say, don't do what I do" 🤣

Reply:
@echlebek Actually what I did worked because I set a slice as the default value, which will cause `viper.GetString` to return an empty value:

https://github.com/sensu/sensu-go/blob/master/agent/cmd/start.go#L238

Definitely not a clean solution though

Reply:
OK. The subscriptions flag has a default value of an empty slice as well. I don't think we can support the canonical way to represent slices, (`--foo a --foo b`) as it would break the CLI... any suggestions?

Reply:
@echlebek Oh, we simply need to use `StringSlice` in pflag, I missed that. So we could simply do:

```
cfg.Subscriptions = viper.GetStringSlice(flagSubscriptions)

[...]

cmd.Flags().StringSlice(flagSubscriptions, viper.GetStringSlice(flagSubscriptions), "comma-delimited list of agent subscriptions")
```

Which would result in the following:

```
$ sensu-agent start --subscriptions foo
slice: [foo]
len: 1

$ sensu-agent start --subscriptions foo,bar
slice: [foo bar]
len: 2

$ sensu-agent start --subscriptions foo,bar --subscriptions qux
slice: [foo bar qux]
len: 3
```

Reply:
Sounds like a plan, thanks!